api/sys_monitor.go

```diff
@@ -0,0 +1,64 @@
+package api
+
+import (
+	"bufio"
+	"context"
+)
+
+func (c *Sys) Monitor(logLevel string, stopCh chan struct{}) (chan string, error) {
+	r := c.c.NewRequest("GET", "/v1/sys/monitor")
+
+	if logLevel == "" {
+		r.Params.Add("log_level", "INFO")
+	} else {
+		r.Params.Add("log_level", logLevel)
+	}
+
+	ctx, cancelFunc := context.WithCancel(context.Background())
+	resp, err := c.c.RawRequestWithContext(ctx, r)
+	if err != nil {
+		return nil, err
+	}
+
+	logCh := make(chan string, 64)
+	go func() {
+		defer resp.Body.Close()
+		scanner := bufio.NewScanner(resp.Body)
+
+		for {
+			select {
+			case <-stopCh:
+				close(logCh)
+				cancelFunc()
+				return
+			case <-ctx.Done():
+				stopCh <- struct{}{}
+				close(logCh)
+				cancelFunc()
+				return
+			default:
+			}
+
+			if scanner.Scan() {
+				// An empty string signals to the caller that
+				// the scan is done, so make sure we only emit
+				// that when the scanner says it's done, not if
+				// we happen to ingest an empty line.
+				if text := scanner.Text(); text != "" {
+					logCh <- text
+				} else {
+					logCh <- " "
```

Comment:
If I understand correctly we only get here when we saw an empty line.  In that case why send a space instead of a newline?

Reply:
This particular piece was copied from either nomad or consul, I forget which.  I'll give your suggestion a try.

Reply:
It might be because the printing of this log line appends a newline already (`c.UI.Info`). I wonder if it's necessary to send anything at all it we receive nothing from `scanner.Text()`.